src/features/todos.ts

```diff
@@ -1,9 +1,28 @@
 import { Todo } from '../types/Todo';
 
-export const actions = {};
+type SetTodosAction = { type: 'todos/SET', payload: Todo[] };
 
-const todosReducer = (): Todo[] => {
-  return [];
+type Action = SetTodosAction;
+type State = Todo[];
```

Comment:
```suggestion
type State = {
  isLoading; boolean;
  hasError: boolean;
  data: Todo[];
};
```
create async action for query todos. You will be able to remove local isLoading and error state

Reply:
removed local isLoading and error state, added these into the async action in App